api/pkg/apis/projectcalico/v3/felixconfig.go

```diff
@@ -450,6 +450,30 @@ type FelixConfigurationSpec struct {
 	// set to false. This reduces the number of metrics reported, reducing Prometheus load. [Default: true]
 	PrometheusWireGuardMetricsEnabled *bool `json:"prometheusWireGuardMetricsEnabled,omitempty"`
 
+	// MetricsTLSEnabled specifies whether TLS encryption is enabled for the /metrics endpoint.
+	// If set to true, the metrics server will only be accessible over HTTPS. Default is false.
+	MetricsTLSEnabled *bool `json:"metricsTLSEnabled,omitempty"`
```

Comment:
I think we can determine this entirely based on whether or not TLS files are provided, avoiding the need for a separate config field - WDYT?